src/ImpactEvaluator.sol

```diff
@@ -62,10 +62,10 @@ contract ImpactEvaluator is AccessControl {
     }
 
     function addMeasurements(string memory cid) public returns (uint) {
+        maybeAdvanceRound();
         uint roundIndex = currentRoundIndex();
         rounds[roundIndex].measurementsCids.push(cid);
         emit MeasurementsAdded(cid, roundIndex);
-        maybeAdvanceRound();
         return roundIndex;
```

Comment:
I find this change problematic for SPARK.

If we associate these measurements with the next round, then the fraud-detection step in the evaluation service will almost certainly discard them as belonging to a wrong round.

Reply:
Got it. We will only associate them with the next round if the next round should already have begun (based on block time). Or is it in general safer to associate with the previous round?

Reply:
Let's discuss, I'm going to land this now since it contains multiple updates